pkg/compactor/bucket_compactor.go

```diff
@@ -598,11 +599,11 @@ func NewBucketCompactorMetrics(blocksMarkedForDeletion prometheus.Counter, reg p
 			Help:        "Total number of blocks that were marked for no-compaction.",
 			ConstLabels: prometheus.Labels{"reason": block.OutOfOrderChunksNoCompactReason},
 		}),
-		blocksMaxTimeDelta: promauto.With(reg).NewHistogram(prometheus.HistogramOpts{
+		blocksMaxTimeDelta: promauto.With(reg).NewHistogramVec(prometheus.HistogramOpts{
 			Name:    "cortex_compactor_block_max_time_delta_seconds",
 			Help:    "Difference between now and the max time of a block being compacted in seconds.",
 			Buckets: prometheus.LinearBuckets(86400, 43200, 8), // 1 to 5 days, in 12 hour intervals
-		}),
+		}, []string{"out_of_order_compaction"}),
```

Comment:
If you accept my suggestion below then, instead of having a boolean, I would consider having something like `block_type` where current values could be something like `standard` and `out_of_order`.